features/ifttt/src/main/java/org/opennms/features/ifttt/IfTttDaemon.java

```diff
@@ -0,0 +1,349 @@
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.core.spring.FileReloadContainer;
+import org.opennms.core.xml.JaxbUtils;
+import org.opennms.features.ifttt.config.IfTttConfig;
+import org.opennms.features.ifttt.config.Trigger;
+import org.opennms.features.ifttt.config.TriggerPackage;
+import org.opennms.features.ifttt.config.TriggerSet;
+import org.opennms.features.ifttt.helper.DefaultVariableNameExpansion;
+import org.opennms.features.ifttt.helper.IfTttTrigger;
+import org.opennms.features.ifttt.helper.VariableNameExpansion;
+import org.opennms.netmgt.dao.api.AlarmDao;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsSeverity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallbackWithoutResult;
+import org.springframework.transaction.support.TransactionOperations;
+
+import com.google.common.base.Strings;
+
+/**
+ * This class represents a daemon for polling alarms in order to generate IFTTT events.
+ */
+public class IfTttDaemon {
+    private static final Logger LOG = LoggerFactory.getLogger(IfTttDaemon.class);
+    /**
+     * the alarm dao instance
+     */
+    private final AlarmDao alarmDao;
+    /**
+     * the transaction template
+     */
+    private final TransactionOperations transactionOperations;
+    /**
+     * file reload container for the ifttt-config.xml file
+     */
+    private final FileReloadContainer<IfTttConfig> m_fileReloadContainer;
+    /**
+     * scheduler for alarm poller
+     */
+    private ScheduledExecutorService m_alarmPoller;
+    /**
+     * the config file instance
+     */
+    private final File ifTttConfigFile = Paths.get(System.getProperty("opennms.home", ""), "etc", "ifttt-config.xml").toFile();
+
+    /**
+     * Constructor to create the daemon instance.
+     *
+     * @param alarmDao              the alarm dao instance to be used
+     * @param transactionOperations the transaction template to be used
+     */
+    public IfTttDaemon(final AlarmDao alarmDao, final TransactionOperations transactionOperations) {
+        this.alarmDao = alarmDao;
+        this.transactionOperations = transactionOperations;
+
+        // initialize the FileReloadContainer instance for the configuration file
+
+        this.m_fileReloadContainer = new FileReloadContainer<>(ifTttConfigFile, (object, resource) -> {
+            LOG.debug("Reloading configuration file {}.", resource.getFilename());
+            return JaxbUtils.unmarshal(IfTttConfig.class, ifTttConfigFile);
```

Comment:
The callback is just for reloading. The FileReloadContainer does not actively watch for file changes. Only when triggered via getObject().